### PR TITLE
[JENKINS-31648] First acceptance test for Folder plugin.

### PR DESCRIPTION
**Pull request by andresrc**
[JENKINS-31648] (https://issues.jenkins-ci.org/browse/JENKINS-31648)

A simple first scenario to start incorporating folder-related tests to the ATH. A `FolderJob` is introduced as part of the process.

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
Very useful if it adds JIRA issue link [JENKINS-31648](https://issues.jenkins-ci.org/browse/JENKINS-31648) to PR description.

File: src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2014 Jesse Glick.
```
Comment (recena) [inline src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java]: Wrong copyright.

File: src/test/java/plugins/FolderPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        // 1 - Create job
+        final FolderJob job = jenkins.jobs.create(FolderJob.class, F01);
+        // 2 - Save job
+        job.save();
+    	// 3 - Go to main page.
+        jenkins.open();
+    	// 4 - Go to job page.
+        job.open();
+        // Ensure the folder name is correct.
+    	MatcherAssert.assertThat(driver, Matchers.hasContent(F01));
```
Comment (recena) [inline src/test/java/plugins/FolderPluginTest.java]: Wrong indentation

File: src/test/java/plugins/FolderPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class FolderPluginTest extends AbstractJUnitTest {
+    /** Test folder name. */
+    private static final String F01 = "F01";
+    
+    /**
+     * First simple test scenario: Folder creation (JENKINS-31648)
+     *
+     * - We create a folder named "F01".
+     * - We check the folder exists and we can enter it.
+     */
```
Comment (recena) [inline src/test/java/plugins/FolderPluginTest.java]: This javadoc won't be valid using JDK8.

File: src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.net.URL;
+
+import com.google.inject.Injector;
+
+/**
+ * Job representing a folder.
+ */
+@Describable("com.cloudbees.hudson.plugins.folder.Folder")
+public class FolderJob extends Job {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java]: I do not think extending `Job` is a good idea for folder. We need a `TopLevelItem` abstraction used by both `Job` and `Folder`.

File: src/test/java/plugins/FolderPluginTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @Test
+    public void createFolder() throws Exception {
```
Comment (olivergondza) [inline src/test/java/plugins/FolderPluginTest.java]: It is my personal thing, though all these comments hurt readability more than help it.

File: src/test/java/plugins/FolderPluginTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * First simple test scenario: Folder creation (JENKINS-31648).
+     * <ol>
+     * <li>We create a folder named "F01".
+     * <li>We check the folder exists and we can enter it.
+     * </ol>
+     */
+    @Test
+    public void createFolder() {
+        // 1 - Create job
```
Comment (olivergondza) [inline src/test/java/plugins/FolderPluginTest.java]: It is my personal thing, though all these comments hurt readability more than it help. Both the method comment and inline comments.

File: src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java (hunk 4 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/test/acceptance/po/FolderJob.java]: Agreed.

**Comment by andresrc** [comment]
Thank you everyone for your comments. I have updated the issue and pushed an additional commit introducing the TopLevelItem abstraction.

**Comment by olivergondza** [comment]
LGTM

**Comment by amuniz** [comment]
:bee:

**Comment by andresrc** [comment]
@reviewbybees done

**Pull request merged**